### ISSUE TITLE
The initialization of learnable parameter t

### ISSUE THREAD

**Issue by zhl98** [zju-vipa/MEAT-TIL#7]
The initial weights of t are independently sampled from distribution Uniform(-γ, γ), where γ > 0 is a hyperparameter.

If the parameter t is initialized to a negative number, there will be a problem in calculating the log?

**Comment by jjunsss** [comment]
@zhl98 
hi !!
I'm wondering about this problem too !! 
Did you solve it??

**Comment by zhl98** [comment]
In some Gumbel Softmax implementations, softmax is used for parameter t first, and then log.
So, log(t) should be updated as log(softmax(t)) .
 The softmax fuction is used for parameter first as shown in the next figure:
![image](https://user-images.githubusercontent.com/51871747/203947996-b969b06f-0c57-4f5b-9427-4382ca4793a9.png)
Please also refer to this page https://pytorch.org/docs/stable/generated/torch.nn.functional.gumbel_softmax.html.
